x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,718 @@
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/node3
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+--
+
+. If necessary, rename the `http.p12` file to match the name of your existing
+certificate for HTTP client communications. For example, `node1-http.p12`.
+
+. [[start-rolling-restart-http,step 5]]On the current node in your cluster where you're updating the keystore,
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1-http.p12`.
++
+IMPORTANT: If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: node1-http.p12
+----
+
+. If your keystore password is changing, add the password for your private key
+to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-http,step 10]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-http>> through
+<<verify-keystore-http>> until you've updated all keystores in your cluster.
+
+[[update-node-certs-different]]
+=== Update security certificates with a different CA
+++++
+<titleabbrev>With a different CA</titleabbrev>
+++++
+If you have to trust a new CA from your organization, or you need to generate a new CA yourself, use this new CA to sign the new node certificates and instruct your nodes to trust the new CA. 
+
+[[node-certs-different-transport]]
+==== Generate a new certificate for the transport layer
+Create a new CA certificate and add it to your existing CA truststore. After
+you finish updating your certificates for all nodes, you can remove the old CA
+certificate from your truststore (but not before!).
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+keystores that are currently in use. You'll use the same names
+for your new keystores.
++
+In this example, the keystore and truststore are using different files.
+Your configuration might use the same file for both the keystore and the 
+truststore. Elastic recommends using separate files. 
```

Comment:
Do we? Why is that?

Reply:
I was talking to an SA and they indicated that using separate files for the keystore and truststore provided an easier mechanism for updating one or the other in case of compromise, and also that the using separate files provides a slight security advantage. That said, we might not explicitly recommend this particular usage. I'll remove the statement about recommending separate files.